storage/innobase/dict/dict0load.cc

```diff
@@ -2456,6 +2456,7 @@ static dict_table_t *dict_load_table_one(const span<const char> &name,
 		ib::error() << "Refusing to load corrupted table "
 			    << table->name;
 evict:
+		fk_tables.clear();
```

Comment:
The modified test `innodb.foreign_key` is not failing for me on 1b9d5cdb83f235f313ae19bab1c6978837a566d7 even if I omit this code change.

I may be mistaken here, but it would seem to me that if there is a larger bunch of tables that are connected via `FOREIGN KEY` constraints, and only one table is causing trouble, we had better only remove those entries that had been added by this invocation of `dict_load_table_one()`. Neither the commit message nor the test case are explaining for which table this added code is expected to matter.

Reply:
Please run it in asan/valgrind build. I will address the comment in the revised patch.